Mage.Sets/src/mage/cards/r/RinAndSeriInseparable.java

```diff
@@ -0,0 +1,119 @@
+package mage.cards.r;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.common.SpellCastControllerTriggeredAbility;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.dynamicvalue.DynamicValue;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.common.CreateTokenEffect;
+import mage.abilities.effects.common.DamageTargetEffect;
+import mage.abilities.effects.common.GainLifeEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.filter.FilterPermanent;
+import mage.filter.FilterSpell;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.permanent.ControllerIsActivePlayerPredicate;
+import mage.game.Game;
+import mage.game.permanent.token.CatToken3;
+import mage.game.permanent.token.DogToken2;
+import mage.target.common.TargetAnyTarget;
+
+import java.util.UUID;
+
+/**
+ *
+ * @author htrajan
+ */
+public final class RinAndSeriInseparable extends CardImpl {
+
+    private static final FilterSpell dogSpellFilter = new FilterSpell("a Dog spell");
+    private static final FilterSpell catSpellFilter = new FilterSpell("a Cat spell");
+
+    private static final FilterPermanent dogPermanentFilter = new FilterCreaturePermanent("dog you control");
+    private static final FilterPermanent catPermanentFilter = new FilterCreaturePermanent("cat you control");
```

Comment:
Wrong filter class -- must be FilterControlledCreaturePermanent, so you don't need `ControllerIsActivePlayerPredicate` (it's for different purpose on opponent's turns).

Reply:
fixed